File: app/src/main/java/com/softaai/designpatterns/java/structural/proxy/ProxyDesignPatternDemo.java
package com.softaai.designpatterns.java.structural.proxy;

/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
public class ProxyDesignPatternDemo {
    public static void main(String[] args){
        Image image = new ProxyImage("test_10mb.jpg");

        // image will be loaded from disk
        image.display();
        System.out.println("");

        // image will not be loaded from disk
        image.display();
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/Proxy/ProxyDesignPatternDemo.kt
package com.softaai.designpatterns.kotlin.structural.Proxy


/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
object ProxyDesignPatternDemo {
    @JvmStatic
    fun main(args : Array<String>){
        var image = ProxyImage("test_10mb.jpg")
        image.display()
        println()
        image.display()
    }
}